docs/MIGRATING.md

```diff
@@ -6,6 +6,17 @@
   list to a list of type `[]string`.
 - The field `CodeInfoResponse.Checksum` is now explicitly marked as
   non-optional. It has always been set to a 32 byte value in the past.
+- All entrypoint functions now return the full result with an `Ok` or `Err`
+  variant instead of just the data inside the `Ok`. This was previously only the
+  case for `IBCPacketReceive`. It is important to note that this means contract
+  errors are no longer returned in the `error` return value. Instead, the `Err`
+  field should be checked for errors.
```

Comment:
The way I like to think about this change is preserving nested results instead of flattening two levels if errors.